examples/bigquery_user_info_updater_tool/cron/bigquery_user_info_updater/updater_tools/user_info_updater.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+from dateutil import tz
+import logging
+
+from google.cloud import bigquery
+
+INITIAL_TIMESTAMP = datetime.datetime(1900, 1, 1, 00, 00, 00, 000,
+                                      tzinfo=tz.tzutc())
+
+
+class UserInfoUpdater(object):
+    """Class for updating Identity data in BigQuery.
+
+    Holds methods for gathering updates to identity data and merging
+    updates into the master identity table.
+
+    Attributes:
+        bq_client(google.cloud.bigquery.client.Client): Client to hold
+            configurations needed for BigQuery API requests.
+        project_id(str): Id of project holding BQ resources.
+        dataset_id(str): Id of dataset holding Identity BQ tables.
+        updates_table_id(str): Id of table holding updates for the identity
+            table.
+        temp_updates_table_id(str): Id of table that deduped update rows will
+            be written to.
+        final_table_id(str): Id of identity table holding golden record for
+            each user.
+
+    """
+
+    def __init__(
```

Comment:
jaketf on Aug 27, 2019 Member
Not necessary but, fyi if you're using python3.7 there's a feature called dataclasses
which abstract away the boiler plate self.foo = foo parts of __init__